dashboard/app/models/pd/application/application_base.rb

```diff
@@ -255,5 +255,10 @@ def total_score
     def include_additional_text(hash, field_name, *options)
       hash[field_name] = answer_with_additional_text hash, field_name, *options
     end
+
+    def school
+      school_id = sanitize_form_data_hash[:school].to_i
+      School.find_by_id(school_id)
```

Comment:
Make sure to handle -1 school id, which means we had a custom school. In that case, school name is specified as text and district will be missing.